src/arviz_plots/plots/ppcplot.py

```diff
@@ -0,0 +1,529 @@
+        plot_collection = PlotCollection.wrap(
+            pp_distribution,
+            backend=backend,
+            **pc_kwargs,
+        )
+    # set reduce_dims by elimination
+    reduce_dims = [
+        dim for dim in pp_distribution.dims if dim not in set(facet_dims).union(set(sample_dims))
+    ]
+
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map.setdefault("predictive", plot_collection.aes_set)
+    # setting aggregate aes_map to `[]` so `overlay` isn't applied for it
+    aes_map.setdefault("aggregate", [])
+    if labeller is None:
+        labeller = BaseLabeller()
+
+    # checking plot_collection wrapped dataset and viz/aes datatrees
+    # print(f"\nplot_collection.data = {plot_collection.data}")
+    # print(f"\nplot_collection.aes = {plot_collection.aes}")
+    # print(f"\nplot_collection.viz = {plot_collection.viz}")
+
+    # setting plot_kwargs_dist defaults (for passing to internal plot_dist calls)
+    plot_kwargs_dist = {
+        key: False
+        for key in ("credible_interval", "point_estimate", "point_estimate_text", "title")
+    }
+    if "remove_axis" in plot_kwargs:
+        plot_kwargs_dist["remove_axis"] = plot_kwargs["remove_axis"]
+
+    # print(f"\n aes_map = {aes_map}")
+
+    # ---------STEP 1 (PPC data)-------------
+    # print(f"\nposterior predictive distri = {pp_distribution!r}")
+
+    # density calculation for observed variables
+    pp_kwargs = copy(plot_kwargs.get("predictive", {}))
+    # print(f"\nreduce_dims = {reduce_dims!r}")
+
+    if pp_kwargs is not False:
+        _, pp_density_aes, _ = filter_aes(plot_collection, aes_map, "predictive", reduce_dims)
+        # print(f"\npp_density_aes = {pp_density_aes}\npp_density_ignore= {pp_density_ignore}")
+
+        # getting first default color from color cycle and picking it
+        pp_default_color = plot_bknd.get_default_aes("color", 1, {})[0]
+        if "color" not in pp_density_aes:
+            pp_kwargs.setdefault("color", pp_default_color)
+
+        if "alpha" not in pp_density_aes:
+            pp_kwargs.setdefault("alpha", 0.2)
+
+        # passing plot_kwargs["predictive"] to plot_kwargs_dist (unlike tracedistplot there are
+        # multiple artists generated via plot_dist in this plot- plot_ppc)
+        plot_kwargs_dist[kind] = pp_kwargs
+
+        # calling plot_dist with plot_collection and customized args
+        pp_dt = DataTree(name="pp_dt", data=pp_distribution)  # has to be converted from a
+        # dataarray to a datatree first before passing to plot_dist
+
+        plot_dist(
+            pp_dt,
+            group="pp_dt",
+            sample_dims=reduce_dims,
+            kind=kind,
+            plot_collection=plot_collection,
+            labeller=labeller,
+            aes_map={
+                kind: value for key, value in aes_map.items() if key == "predictive"
+            },  # aes_map[kind] is set to "predictive" aes_map
```

Comment:
```suggestion
            aes_map={kind: aes_map.get("predictive", {})},
```

Not sure if you were getting errors when "predictive" key wasn't in `aes_map` but this should not be a loop, we are always selecting only one key and we know which one beforehand. Same for the other similar situations.